crates/mipsy_interactive/src/interactive/helper.rs

```diff
@@ -33,15 +53,47 @@ impl Completer for MyHelper {
         pos: usize,
         ctx: &Context<'_>,
     ) -> Result<(usize, Vec<Pair>), ReadlineError> {
-        self.completer.complete(line, pos, ctx)
+        self.completer.complete(line, pos, ctx).and_then(|x| {
+            // will be 0 for first arg. kinda a hacky way to check if we should match
+            // against files or not but thats okay.
+            // TODO: something like `match self.arg_type() { label => , file => number => }`
+            Ok(if x.0 == 0 {
```

Comment:
comments

Reply:
this is also a bit hacky in general since the way we read history at this point is to collect an iterator since the iterator accesses hidden fields